File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        n = len(nums) 
        mpp = collections.defaultdict(int)
        preSum = 0
        cnt = 0
        mpp[0] = 1 # Setting 0 in the map.
        for i in range(n):
            preSum += nums[i]
            remove = preSum - k
            cnt += mpp[remove]
            mpp[preSum] += 1
        return cnt
        

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var subarraySum = function(nums, k) {
  let res=0
	let presum=0
	const dict1={0:1}
	for(let num of nums){
	  presum =presum+num
	  if (presum-k in dict1){
	    res =res+dict1[presum-k]
	  }
	  if (presum in dict1){
	      dict1[presum] ++
	    }else{
      dict1[presum]=1
	      
	    }       
	     
	  
	}

    return res
};